include/event_dbus_monitor.hpp

```diff
@@ -64,29 +130,221 @@ inline void VMIIPChange(sdbusplus::message::message& msg)
+
+        if (objPath.find("/eth0") != std::string::npos)
+        {
+            isIPEnabledOnIntfEth0 = *propValue;
+            intf = "eth0";
+        }
+        else if (objPath.find("/eth1") != std::string::npos)
+        {
+            isIPEnabledOnIntfEth1 = *propValue;
+            intf = "eth1";
+        }
+
+        if (((intf == "eth0") && (methodOnIntf0 == ipMethodStatic)) ||
+            ((intf == "eth1") && (methodOnIntf1 == ipMethodStatic)))
+        {
+            // send event only if it is static
+            // as the enabled property will be set to true
+            // only after the ip configuration is applied
+            // on the hypervisor networkd dbus object.
+            // In case of dhcp configuration, the enabled
+            // property will be set to true before the host
+            // sends down the dhcp ip details, in that case
+            // events will be sent for each property change
+            // on the dbus object.
+
+            std::string origin =
+                "/redfish/v1/Systems/hypervisor/EthernetInterfaces/" + intf;
+            BMCWEB_LOG_DEBUG << "Pushing the VMI IP property change event for "
+                                "static IP configuration on " +
+                                    intf;
+            sendEventOnEthIntf(origin);
+            return;
+        }
+    }
+}
+
+inline void VMIIPPropertyChange(sdbusplus::message::message& msg)
+{
+    if (msg.is_method_error())
+    {
+        BMCWEB_LOG_ERROR << "BMC Hypervisor IP properties changed Signal error";
+        return;
+    }
+
+    std::string objPath = msg.get_path();
+
+    if ((objPath !=
+         "/xyz/openbmc_project/network/hypervisor/eth0/ipv4/addr0") &&
+        (objPath != "/xyz/openbmc_project/network/hypervisor/eth1/ipv4/addr0"))
+    {
+        return;
+    }
+
+    std::string infName;
+    if (objPath.find("/eth0") != std::string::npos)
+    {
+        infName = "eth0";
+    }
+    else if (objPath.find("/eth1") != std::string::npos)
+    {
+        infName = "eth1";
+    }
+
+    boost::container::flat_map<std::string, std::variant<std::string, uint8_t>>
+        values;
+    std::string objName;
+    msg.read(objName, values);
+
+    bool enabledValue;
+    if (infName == "eth0")
+    {
+        enabledValue = isIPEnabledOnIntfEth0;
+    }
+    else if (infName == "eth1")
+    {
+        enabledValue = isIPEnabledOnIntfEth1;
+    }
+
+    std::string origin =
+        "/redfish/v1/Systems/hypervisor/EthernetInterfaces/" + infName;
+    if (((infName == "eth0") && (methodOnIntf0 == ipMethodStatic)) ||
+        ((infName == "eth1") && (methodOnIntf1 == ipMethodStatic)))
+    {
+        // Change if enabled property by pldm will
+        // take care of sending the event for static
+        // configuration
+        return;
+    }
+
+    auto find = values.find("Address");
+    if (find != values.end())
+    {
+        const std::string propValue = std::get<std::string>(find->second);
```

Comment:
This throws doesn't it ? 
std::get_if and then check null ?

Reply:
done